oasislmf/pytools/pla/streams.py

```diff
@@ -70,13 +69,18 @@ def __init__(self, items_amps, plafactors, default_factor):
         self.default_factor = default_factor
         self.out_byte_mv = np.empty(PIPE_CAPACITY, dtype='b')
         self.out_cursor = np.empty(1, dtype='i4')
+
+        self.event_id = 0
+        self.item_id = 0
+
         self.logger = logger
 
     def read_buffer(self, byte_mv, cursor, valid_buff, event_id, item_id):
-        return read_buffer(
-            byte_mv, cursor, valid_buff, event_id, item_id,
+        cursor, self.event_id, self.item_id, yield_event = read_buffer(
+            byte_mv, cursor, valid_buff, self.event_id, self.item_id,
             self.items_amps, self.plafactors, self.default_factor, self.out_byte_mv, self.out_cursor
         )
+        return cursor, self.event_id, self.item_id, yield_event
```

Comment:
It looks fine to me @sstruzik, but just a question: are `event_id` and `item_id` are unused variables in this method?

Reply:
yes, but it is in the generic function signature read_buffer(self, byte_mv, cursor, valid_buff, event_id, item_id)
That's why it is there.
Normally each type a event is fully read, the read return the event but in pla we don't need that we can read the full buffer update the loss as we read and return the full buffer regardless of the item. however we need to keep track of event_id and item_id that are normally reset to 0 if we tell the upper function to yield the event